### PR TITLE
CIA-24 Allow customizing tippy via prop

### PR DESCRIPTION
**Pull request by hoangnd25**

### CODE REVIEW

File: src/components/Popover/index.tsx [hunk 1]
```diff
@@ -54,6 +56,7 @@ export default class Popover extends React.Component<Props, any> {
             </PopoverDisplay>
           }
           position={preferRight ? 'bottom-end' : 'bottom-start'}
+          {...tippyProps || {}}
```
Comment (zediah) [inline src/components/Popover/index.tsx]: Set an empty object in default props instead :)

File: src/components/Popover/index.tsx [hunk 2]
```diff
@@ -11,6 +11,7 @@ type Props = {
   tooltip?: boolean;
   light?: boolean;
   className?: string;
+  tippyProps?: object;
```
Comment (zediah) [inline src/components/Popover/index.tsx]: Is tippy a known library? Does it have a definition file? Can you give a better type than 'object'?

File: src/components/Popover/index.tsx (hunk 1 above)
Comment (zediah) [reply src/components/Popover/index.tsx]: Also, please put spread props like this at the -start- of the component. If you put it at the end, people can override (accidently or on purpose) already set props on the object.

File: src/components/Popover/index.tsx [hunk 3]
```diff
@@ -43,7 +49,7 @@ export default class Popover extends React.Component<Props, any> {
           delay={100}
           hideDelay={300}
           interactiveBorder={16}
-          trigger={triggerOnClick ? 'click' : 'mouseenter'}
+          trigger={trigger || (triggerOnClick ? 'click' : 'mouseenter')}
```
Comment (aoimasu) [inline src/components/Popover/index.tsx]: if we gonna use react-tippy's props, i think it's best to remove triggerOnClick and use trigger.

File: src/components/Popover/index.tsx [hunk 4]
```diff
@@ -53,7 +59,8 @@ export default class Popover extends React.Component<Props, any> {
               {content}
             </PopoverDisplay>
           }
-          position={preferRight ? 'bottom-end' : 'bottom-start'}
+          position={position || (preferRight ? 'bottom-end' : 'bottom-start')}
```
Comment (aoimasu) [inline src/components/Popover/index.tsx]: same as above for position

**Pull request merged**